t/31-sshSerial.t

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/perl
+
+use Test::Most;
+use FindBin '$Bin';
+use lib "$Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '20';
+use Test::MockObject;
+use Test::Warnings ':report_warnings';
+use Test::Output;
+use Net::SSH2 'LIBSSH2_ERROR_EAGAIN';
+
+use consoles::sshSerial;
+
+my $eagain = [
+    LIBSSH2_ERROR_EAGAIN,
+    'LIBSSH2_ERROR_EAGAIN',
+    'Operation would block'
+];
+
+my $mock_backend = Test::MockObject->new();
+my $mock_ssh     = Test::MockObject->new();
+my $mock_channel = Test::MockObject->new();
+
+$mock_ssh->{error} = undef;
+
+$mock_channel->{write_limits} = [];
+$mock_channel->{write_buffer} = '';
+$mock_channel->{read_queue}   = [];
+$mock_channel->{blocking}     = 1;
+
+$mock_channel->mock(pty => sub {
+        return 1;
+});
+
+$mock_channel->mock(ext_data => sub {
+        return 1;
+});
+
+$mock_channel->mock(shell => sub {
+        return 1;
+});
+
+$mock_channel->mock(blocking => sub {
+        my $self = shift;
+        my $arg  = shift;
+
+        $self->{blocking} = $arg if defined($arg);
+        return $self->{blocking};
+});
```

Comment:
```suggestion
$mock_channel->mock(blocking => sub {
        my ($self, $arg) = @_;
        $self->{blocking} = $arg if defined($arg);
});
```

Reply:
Fixed argument declaration. But dropping the `return` statement will break `ssh_screen::do_read()`.